volume/parser.go

```diff
@@ -0,0 +1,43 @@
+package volume
+
+import (
+	"runtime"
+
+	"github.com/docker/docker/api/types/mount"
+)
+
+const (
+	// PlatformLinux is the same as runtime.GOOS on linux
+	PlatformLinux = "linux"
+	// PlatformWindows is the same as runtime.GOOS on windows
+	PlatformWindows = "windows"
+)
+
+// Parser represents a platform specific parser for mount expressions
+type Parser interface {
+	ParseMountRaw(raw, volumeDriver string) (*MountPoint, error)
+	ParseMountSpec(cfg mount.Mount) (*MountPoint, error)
+	ParseVolumesFrom(spec string) (string, string, error)
+	DefaultPropagationMode() mount.Propagation
+	ConvertTmpfsOptions(opt *mount.TmpfsOptions, readOnly bool) (string, error)
+	DefaultCopyMode() bool
+	IsVolumeNameValid(name string) (bool, error)
```

Comment:
Yes I agree, I'll just have to check on error type instead of the combination of both bool and error (which is confusing)